jtsync/jtsync.rb

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env ruby
+#
+# Copyright 2016, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+ENV["CURRENT_GEMFILE"] ||= File.expand_path("Gemfile", __FILE__)
+
+require "bundler"
+Bundler.setup(:default)
+require "trello"
+
+
+class Failed < StandardError
+end
+
+
+def board
+  @trello_board ||= if ENV.has_key? "TRELLO_BOARD"
+                      Trello::Board.find(ENV["TRELLO_BOARD"])
+                    else
+                      Trello::Board.find("ywSwlQpZ")
+                    end
+end
+
+
+def state_from_env
+  case ENV["JOB_STATUS"]
+  when "0" then 'successful'
+  else 'failed'
+  end
+end
+
+
+# Handling different naming conventions between opensuse/suse
+# and matrix/normal job
+#
+# returns tuple: (list_name, job_name)
+def job_from_env
+  # is this a ci.opensuse.org matrix job? (Juno)
+  if ENV.has_key? "openstack_project"
+    return ["OpenStack", "#{ENV["JOB_NAME"].gsub("openstack-", "")}: #{ENV["openstack_project"]}"]
```

Comment:
This kind of parsing within the script hides the dependency of this script to these variables. Some of them are Jenkins variables, but others are variables that are used within the jenkins job. So if the job changes, it is not visible that this script needs to be changed as well.

So my recommendation is to switch the call of this script to something like:
```
jtsync.rb openstack $openstack_project $other_variable
# other job calls eg.
jtsync.rb trackupstream $COMPONENT $this_variable $that_variable
```
By this it is obvious that the script needs the values from these variables and if the job is changed this call will also be adapted. It also reduces the amount of parsing and if-else cases in the jtsync script.